File: Amazon/TwoSum.go
package amazon

import (
)

// TwoSum returns the indices of the two numbers such that they add up to a specific target.
func TwoSum(nums []int, target int) []int {
	// Create a map to store the index of each number
	indexMap := make(map[int]int)

	// Iterate through the numbers
	for i, num := range nums {
		// Calculate the complement
		complement := target - num

		// Check if the complement is in the map
		if index, ok := indexMap[complement]; ok {
			// If it is, return the indices
			return []int{index, i}
		}

		// Otherwise, add the number to the map
		indexMap[num] = i
	}

	// If no solution is found, return an empty slice
	return []int{}
}

File: Amazon/TwoSum.cs
namespace Amazon
{
    public class TwoSum
    {
        public int[] GetTwoSum(int[] nums, int target)
        {
            var dict = new Dictionary<int, int>();
            for (int i = 0; i < nums.Length; i++)
            {
                var complement = target - nums[i];
                if (dict.ContainsKey(complement))
                {
                    return new int[] { dict[complement], i };
                }
                dict[nums[i]] = i;
            }
            return new int[] { -1, -1 };
        }
    }
}